### ISSUE TITLE
Add unit support for image block height and width

### ISSUE THREAD

**Issue by richtabor** [WordPress/gutenberg#55059]
In an effort to consolidate aspect ratio across blocks, let's add the unit control functionality that the Featured Image block supports for height and width, to the Image block.  Note that aspect ratio should be maintained, regardless of units. 

The image block only supports `px` currently:

![CleanShot 2023-10-04 at 14 36 08](https://github.com/WordPress/gutenberg/assets/1813435/c28f3141-8fe5-4ff3-8e3c-25203267e0a7)

As seen in the featured image block: 

![CleanShot 2023-10-04 at 14 34 20](https://github.com/WordPress/gutenberg/assets/1813435/957e6cd1-958a-4ce0-b1eb-52dba8a832d8)

**Comment by bgardner** [comment]
@richtabor For consistency sake, should we mirror the order of Height / Width across the two blocks? Image is Width / Height while Featured Image is Height / Width. (If so, my vote is for the former)

**Comment by richtabor** [comment]
> @richtabor For consistency sake, should we mirror the order of Height / Width across the two blocks? Image is Width / Height while Featured Image is Height / Width. (If so, my vote is for the former)

Certainly. I have this issue to consolidate into a proper component, which would make these 1:1 across the board. https://github.com/WordPress/gutenberg/issues/47962

**Comment by gsallman** [comment]
Please, **this needs to be implemented as soon as possible since the current implementation is proving to be a backward compatibility nightmare**. The previous functionality used %-ages, which scaled responsively.  The new implementation does not scale responsively, and if you open an old post to make a minor edit, any images in the post using the old % sizes are updated and given absolute sizes, which do not scale responsively. :(

In short, by using absolute scaling units (px), adding images to a post, adjusting their sizes, and having them responsively scale on various devices is impossible.

**Comment by brettsmason** [comment]
I'd love to see this implemented too. Having recently discovered its not possible to extend the list of build-in aspect ratios (unless anyone can could explain how?), this would be the next best thing. Or a way to enter a custom aspect ratio.

**Comment by benoitchantre** [comment]
@richtabor Adding support for more units as proposed would allow layouts with two columns using wide with and an image stretched to the edge without CSS.

**Comment by TheRadicalDreamer** [comment]
Also, it should have min and max units, because i have too  many issues with responsive designs when i set images on certain sizes.

**Comment by tonnysantana2004** [comment]
I need this feature, thanks!
